### ISSUE TITLE
Catalog Price Rule applied to Customer Group wrong price value on minicart, cart after adding to cart.

### ISSUE THREAD

**Issue by tuyennn** [magento/magento2#23986]
### Preconditions (*)

1. Magento 2.1.17 & 2.2.7

### Steps to reproduce (*)

1. Create New customer group, assign new group to a customer
2. Create new Catalog Price Rule with discount and specific applied for the new customer group
3. Login on new customer account and saw the product price on product view, minicart and cart after adding this to cart.
4. If I land on checkout -> the price was correct(But do not land on checkout at first)

### Expected result (*)

1. Discount price applied to product in product view will exact the same with minicart price and cart price

### Actual result (*)

1. Price of product on minicart and cart was till base price not the discounted price.

### Investigate
1. From
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Quote/Observer/Frontend/Quote/Address/CollectTotalsObserver.php#L118-L138

It seem like the customer groupId was always set to 1 while it's getting **getDefaultGroup**
```
        $groupId = null;
        if (empty($customerVatNumber) || false == $this->customerVat->isCountryInEU($customerCountryCode)) {
            $groupId = $customer->getId() ? $this->groupManagement->getDefaultGroup(
                $storeId
            )->getId() : $this->groupManagement->getNotLoggedInGroup()->getId();
        } else {
            // Magento always has to emulate group even if customer uses default billing/shipping address
            $groupId = $this->customerVat->getCustomerGroupIdBasedOnVatNumber(
                $customerCountryCode,
                $this->vatValidator->validate($address, $storeId),
                $storeId
            );
        }
        if ($groupId !== null) {
            $address->setPrevQuoteCustomerGroupId($quote->getCustomerGroupId());
            $quote->setCustomerGroupId($groupId);
            $this->customerSession->setCustomerGroupId($groupId);
            $customer->setGroupId($groupId);
            $quote->setCustomer($customer);
        }
```

And then on 
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Quote/Model/Quote/Address/Total/Subtotal.php#L93

```
$product->setCustomerGroupId($quoteItem->getQuote()->getCustomerGroupId());
```
After collecting the price on minicart and cart product price should be always wrong here.

**Comment by tuyennn** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @tuyennn. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @tuyennn, here is your [Magento instance](https://i-23986-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23986-2-3-develop.instances.magento-community.engineering/admin](https://i-23986-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by tuyennn** [comment]
Closed due cannot reproduce on 2.3-develop instance
